features/macros.go

```diff
@@ -0,0 +1,206 @@
+package features
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/sys"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+func init() {
+	macros.macroTypes = make(map[MacroType]error, maxMacroType)
+}
+
+var (
+	macros macroCache
+)
+
+type macroCache struct {
+	sync.Mutex
+	macroTypes map[MacroType]error
+}
+
+type MacroType uint32
+
+// Max returns the latest supported MacroType.
+func (_ MacroType) Max() MacroType {
+	return maxMacroType - 1
+}
+
+const (
+	// LargeInsn support introduced in
+	// commit c04c0d2b968ac45d6ef020316808ef6c82325a82
+	LargeInsn MacroType = iota
```

Comment:
The name is inspired by `bpftool feature probe macros` as this PR covers the very same probes.

https://github.com/torvalds/linux/blob/112450df61b7373529b0fe4c122ad13b89d80a8a/tools/bpf/bpftool/feature.c#L860-L871

Spliting it up and having dedicated functions, like `func HaveBoundedLoops() (error, bool){..}`, is also possible. Would you prefer this?

Reply:
I guess it's called `macros` because that's what the command outputs? The name doesn't make much sense in this context imo. Maybe `misc` is better.

I'd prefer free functions, with return type `(error)` and the same semantics as the other free functions though.

Reply:
applied your suggestions and renamed `macros` to `misc`.